fs.go

```diff
@@ -899,6 +923,70 @@ func (fsys *DeepFS) context() context.Context {
 	return context.Background()
 }
 
+// deepWalkDir recursively descends a, calling walkDirFn.
+func deepWalkDir(fsys *DeepFS, name string, d fs.DirEntry, walkDirFn fs.WalkDirFunc) error {
+	if err := walkDirFn(name, d, nil); err != nil || !d.IsDir() {
+		if err == fs.SkipDir && d.IsDir() {
+			// Successfully skipped directory.
+			err = nil
+		}
+		return err
+	}
+
+	dirs, err := fs.ReadDir(fsys, name)
+	if err != nil {
+		// Second call, to report ReadDir error.
+		err = walkDirFn(name, d, err)
+		if err != nil {
+			if err == fs.SkipDir && d.IsDir() {
+				err = nil
+			}
+			return err
+		}
+	}
+
+	for _, d1 := range dirs {
+		var name1 string
+		if len(fsys.innerFSSeparator) > 0 && name == fsys.Root {
+			name1 = name + fsys.innerFSSeparator + d1.Name()
```

Comment:
TODO:

```suggestion

	useInnerSeparator := len(fsys.innerFSSeparator) > 0 && name == fsys.Root
	for _, d1 := range dirs {
		var name1 string
		if useInnerSeparator {
			name1 = name + fsys.innerFSSeparator + d1.Name()
```

No need to preform that same check over and over again for each dir entry.